src/ServiceControl.RavenDB/EmbeddedDatabase.cs

```diff
@@ -162,23 +162,70 @@ public async Task<IDocumentStore> Connect(CancellationToken cancellationToken)
 
         public async Task DeleteDatabase(string dbName)
         {
-            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName) { SkipCreatingDatabase = true });
+            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName)
+            {
+                SkipCreatingDatabase = true
+            });
             await store.Maintenance.Server.SendAsync(new DeleteDatabasesOperation(dbName, true));
         }
 
+        public async Task Stop(CancellationToken cancellationToken)
+        {
+            Logger.Debug("Stopping RavenDB server");
+            EmbeddedServer.Instance.ServerProcessExited -= OnServerProcessExited;
+
+            await shutdownTokenSource.CancelAsync();
+
+            // TODO: await EmbeddedServer.Instance.StopAsync(cancellationToken);
+
+            var processId = await EmbeddedServer.Instance.GetServerProcessIdAsync(cancellationToken);
+
+            var waitForCancellationTask = Task.Delay(Timeout.Infinite, cancellationToken);
+            var firstTask = await Task.WhenAny(
+                Task.Run(() => EmbeddedServer.Instance.Dispose(), cancellationToken),
+                waitForCancellationTask
+            );
+
+            if (firstTask == waitForCancellationTask)
+            {
+                try
+                {
+                    Logger.Warn("Killing RavenDB child process because host cancelled");
+                    var ravenChildProcess = Process.GetProcessById(processId);
+                    ravenChildProcess.Kill(entireProcessTree: true);
+                }
+                catch (Exception e)
+                {
+                    Logger.Error("Killing RavenDB child process failed", e);
+                }
+            }
+
+            EmbeddedServer.Instance = null!;
+            Logger.Debug("Stopped RavenDB server");
+        }
+
         public void Dispose()
         {
             if (disposed)
             {
                 return;
             }
 
-            EmbeddedServer.Instance.ServerProcessExited -= OnServerProcessExited;
+            if (EmbeddedServer.Instance != null)
+            {
+                EmbeddedServer.Instance.ServerProcessExited -= OnServerProcessExited;
+            }
 
             shutdownTokenSource.Cancel();
-            Logger.Debug("Disposing RavenDB server");
-            EmbeddedServer.Instance.Dispose();
-            Logger.Debug("Dispose RavenDB server");
+
+            if (EmbeddedServer.Instance != null)
+            {
+                serverOptions.GracefulShutdownTimeout = TimeSpan.Zero;
```

Comment:
Should we add a comment here to explain how we are using the "backdoor" to be able to adjust the shutdown timeout?